src/Symfony/Component/HttpKernel/Profiler/FileProfilerStorage.php

```diff
@@ -194,11 +196,55 @@ public function write(Profile $profile): bool
                 $profile->getStatusCode(),
             ]);
             fclose($file);
+
+            $this->removeExpiredProfiles($indexFilename);
         }
 
         return true;
     }
 
+    protected function removeExpiredProfiles(string $indexFilename)
```

Comment:
```suggestion
    private function removeExpiredProfiles(string $indexFilename)
```

I don't think we want to make an extension point.